control-plane/subcommand/flags/http.go

```diff
@@ -161,3 +169,43 @@ func (s *StringValue) String() string {
 	}
 	return current
 }
+
+// DurationValue provides a flag value that's aware if it has been set.
+type DurationValue struct {
```

Comment:
How are these DurationValue functions being used/meant to be used? Was it because we didn't have any duration related flags before and all duration flags need to implement these?

Reply:
Yes, the StringValue was copied from this same file from consul core: https://github.com/hashicorp/consul/blob/21e855d860865f1472b3fdcb64a916e92bc83a9f/command/flags/config.go#L91 and there is a comment at the top of this file from a while ago about why we have copied them.

I originally had this an IntValue but ultimately it is going to become a duration and we already have some places in our helm values where you can configure duration like "5s", "500ms", etc., so I changed it to make it consistent and more flexible.